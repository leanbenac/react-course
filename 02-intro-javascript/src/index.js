const activo = true;

// let mensaje = '';

//  if ( !activo ) {
//      mensaje = 'activo';

//  } else {
//      mensaje = 'inactivo';
//  };
const mensaje = ( !activo ) ? 'activo' : 'inactivo';


 console.log(mensaje);