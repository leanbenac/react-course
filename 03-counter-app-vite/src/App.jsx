import React from "react";
import PropTypes from 'prop-types'


const App = ({ title, subTitle, name }) => {



  return (
    <>
      <h2>{ title }</h2>
      <p>{ subTitle }</p>
      <p>{  name }</p>
    </>
  );
};


export default App;

App.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.number,
}

App.defaultProps = {
    name: 'Leandro Benac',
    subTitle: 19239,
    title: 'Benac',
}