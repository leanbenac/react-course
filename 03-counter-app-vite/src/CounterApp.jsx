import PropTypes from "prop-types";
import { useState } from "react";

const CounterApp = ({ value }) => {
  console.log('render');
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter(counter + 1);
  };

  const handleSubtract = () => {
    setCounter((c) => c - 1);
  };

  const handleReset = () => setCounter(value);
  
  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleSubtract}>-1</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
};

export default CounterApp;

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
