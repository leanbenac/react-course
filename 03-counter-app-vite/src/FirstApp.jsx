import React from "react";
import PropTypes from 'prop-types'


export const FirstApp = ({ title, subTitle, name }) => {

  return (
    <>
      <h1>{ title }</h1>
      <p>{ subTitle }</p>
      <p>{  name }</p>
    </>
  );
};




FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.number,
}

FirstApp.defaultProps = {
    name: 'Leandro Benac',
    subTitle: 19239,
    // title: 'Benac',
}