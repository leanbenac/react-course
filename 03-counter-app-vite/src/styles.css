html, body {
    background-color: cadetblue;
    font-family: 'Courier New', Courier, monospace;
    font-size: 2rem;
    color: azure;
}
button {
    padding: 15px;
    font-size: 1.3rem;
}