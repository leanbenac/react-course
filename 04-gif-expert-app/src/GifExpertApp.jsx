import React from 'react'

const GifExpertApp = () => {
    return (
        <>
            {/* {titulo} */}
            <h1>GifExpertApp</h1>

            {/* {input} */}

            {/* {Listado gif} */}
            {/* {gitItem} */}
        </>
    )
}

export default GifExpertApp
