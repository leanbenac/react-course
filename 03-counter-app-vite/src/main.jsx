import React from "react";
import ReactDOM from "react-dom/client";
import { FirstApp } from './FirstApp'
import CounterApp from './CounterApp'
import './styles.css'

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <CounterApp  value={ 1 } /> */}
    <FirstApp title="Hola, Soy Goku" subTitle="soy un subtitulo" />
  </React.StrictMode>
);
 