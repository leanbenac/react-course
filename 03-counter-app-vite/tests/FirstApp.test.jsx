import { render } from "@testing-library/react";
import { FirstApp } from '../src/FirstApp';


describe('Pruebas en <FirstApp />', () => {

    test('Debe hacer match con el snapshot', () => {
        //renderiza el componnente en memoria 
        const title = 'Hola, Soy Goku';
        const { container } = render ( <FirstApp title= { title }/> )

        expect( container ).toMatchSnapshot();

    })

    test('Debe de mostrar el title en un h1', () => {

        const title = 'Hola, Soy Goku';
        const { container, getByText  } = render ( <FirstApp title= { title }/> )
        expect( getByText(title) ).toBeTruthy()

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain( title )

    })

});

