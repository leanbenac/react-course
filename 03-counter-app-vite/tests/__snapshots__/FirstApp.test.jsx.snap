// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, Soy Goku
  </h1>
  <p>
    19239
  </p>
  <p>
    Leandro Benac
  </p>
</div>
`;
